syncer/ddl.go

```diff
@@ -113,26 +115,55 @@ func (s *Syncer) parseDDLSQL(sql string, p *parser.Parser, schema string) (resul
 	}
 }
 
-// resolveDDLSQL do two things
-// * it splits multiple operations in one DDL statement into multiple DDL statements
-// * try to apply online ddl by given online
+// splitAndFilterDDL will split multi-schema change DDL into multiple one schema change DDL due to TiDB's limitation.
+// the returned DDLs is split from `stmt`, or DDLs that online DDL component has saved before (which means the input
+// `stmt` is a RENAME statement which often generated by online DDL tools).
 // return @spilt sqls, @online ddl table names, @error.
-func (s *Syncer) resolveDDLSQL(tctx *tcontext.Context, p *parser.Parser, stmt ast.StmtNode, schema string) (sqls []string, tables map[string]*filter.Table, err error) {
+func (s *Syncer) splitAndFilterDDL(
+	ec eventContext,
+	p *parser.Parser,
+	stmt ast.StmtNode,
+	schema string,
+) (sqls []string, tables map[string]*filter.Table, err error) {
 	sqls, err = parserpkg.SplitDDL(stmt, schema)
 	if err != nil {
 		return nil, nil, err
 	}
-	if s.onlineDDL == nil {
-		return sqls, nil, nil
-	}
 
 	statements := make([]string, 0, len(sqls))
 	tables = make(map[string]*filter.Table)
 	for _, sql := range sqls {
+		stmt2, err2 := p.ParseOneStmt(sql, "", "")
+		if err2 != nil {
+			return nil, nil, terror.Annotatef(terror.ErrSyncerUnitParseStmt.New(err2.Error()), "ddl %s", sql)
+		}
+
+		tableNames, err2 := parserpkg.FetchDDLTableNames(schema, stmt2)
+		if err2 != nil {
+			return nil, nil, err2
+		}
+
+		// get real tableNames before apply block-allow list
+		if s.onlineDDL != nil {
+			for _, names := range tableNames {
+				names.Name = s.onlineDDL.RealName(names.Name)
+			}
+		}
+
+		ignore, err2 := s.skipQuery(tableNames, stmt2, sql)
```

Comment:
if we skipQuery here, do we need skipQuery in [parseDDLSQL](https://github.com/pingcap/dm/blob/d2a1e30cb2588066d86e6830979fa03e1df8f8af/syncer/ddl.go#L36)

but the `stmt` here is from parseDDLSQL and some users' sql will be rewritten and I'm not sure how to refine the skipQuery logic here.